src/commands/tag.js

```diff
@@ -0,0 +1,114 @@
+import { writeObject } from './writeObject'
+import { readObject } from './readObject'
+import { writeRef } from './writeRef'
+import { config } from './config'
+import { GitRefManager } from '../managers/GitRefManager'
+import { E, GitError } from '../models/GitError.js'
+import { FileSystem } from '../models/FileSystem.js'
+import { join } from '../utils/join.js'
+import { cores } from '../utils/plugins.js'
+import { GitAnnotatedTag } from '../models/GitAnnotatedTag'
+
+/**
+ * Create a lightweight or annotated tag.
+ *
+ * @link https://isomorphic-git.github.io/docs/tag.html
+ */
+export async function tag ({
+  core = 'default',
+  dir,
+  gitdir = join(dir, '.git'),
+  fs: _fs = cores.get(core).get('fs'),
+  name,
+  annotated = undefined,
```

Comment:
I'm torn between the pros of `tag({ annotated: true })` being analogous to `git tag -a` and the perf benefit of splitting `tag` and `annotatedTag` into two separate functions.
If we made them separate functions, then when users `import { tag } from 'isomorphic-git'` they wouldn't have to import the GitAnnotatedTag model at all.

Reply:
I think it's reasonable to separate and no need to make it analogous going as far as sacrificing performance, especially regarding this matter – lightweight tag refs and annotated tag objects are totally different things internally.